### ISSUE TITLE
macos: espanso 2.3.0 does not start from the app bundle

### ISSUE THREAD

**Issue by dim882** [federico-terzi/espanso#2444]
### Description

The wizard never allows me to complete the setup and start the app.

### Steps to reproduce

I start the app, then I get the dialog telling me to enable accessibility privileges, which I do. The wizard doesn't notice that I've enabled it and it doesn't have any way to say it's done. So the only thing to do is close that dialog, which quits the app. ☹️ 

### Expected behavior

I expect the wizard to allow me to complete the setup and start the app.

### Your environment

- Public source code:
- Environment name and version (e.g., Rust 1.59.0):
- Operating system and version (e.g., Ubuntu 20.04.2 LTS): Mac 15.6.1
- Espanso version (e.g., 2.1.4-beta): 2.2.1

### Self-service

- [ ] I'd be willing to fix this bug myself.

**Comment by FatRabidSquirrel** [comment]
I also have this issue. I had a working version of Espanso, but then after a MacOS update it stopped working. I tried completely uninstaling and reinstalling via brew, no dice. Same issue - opening Espanso has it show a window saying that I need to add Espanso to Accessiblity settings, but when I open that settings page, Espanso's slider is already enabled. Closing that settings page, Espanso closes and never starts. Triggers that are in my Espanso files do nothing.

Currently running MacOS Sequoia 15.6.1 on an M1 MacBook Pro. 
Espanso version is 2.2.1 installed via brew.

**Comment by juliankastillo** [comment]
Same issue. I´m using a Mac Book Air and I can't complete the installation. I was a PC user and I never had an issue with the app.

**Comment by bradydan** [comment]
I've just downloaded it and tried to install it on macOS 15.5 (24F74). Firstly, the OS did not want it to install, but I bypassed that. However, when I double-click the application, it does not open. Nothing happens.

**Comment by nadesco** [comment]
I updated to 2.2.5 but now can't get past the _Enable Accessibility_ popup. I tried toggling Espanso on/off in both Accessibility and Automation in System Settings but to no avail. A couple of reboots didn't help either.
MacOS Sequoia 15.7, Mac mini M2.

**Comment by nasrelasri** [comment]
Same issue using MacOs Tahoe on an M1 machine: opening the app does nothing at all.

I downloaded the installer from the official website (v2.2.5) and moved the app into the Applications folder.

**Comment by RobATX** [comment]
Same here. I double click it and there's a flash then nothing. Doesn't load. No crashes in Console.

**Comment by usernotnull** [comment]
+1 on windows 11.
Issue started with version 2.2.5, when run, command window flashes and nothing happens. No error output.

**Comment by toxfu** [comment]
> +1 on windows 11. Issue started with version 2.2.5, when run, command window flashes and nothing happens. No error output.

Same problem

**Comment by otaconix** [comment]
EDIT: the following is _only_ for MacOS users.

Potential workaround: remove Espanso from the list of applications for which the _Accessibility_ permission is/isn't given, then reopen Espanso.

<img width="713" height="619" alt="Image" src="https://github.com/user-attachments/assets/21aca03b-ee6f-422b-8019-f60a181fb5ae" />

This seems to have fixed it for me.

**Comment by usernotnull** [comment]
The issue (on win11) has been fixed on #2478 V2.2.7
You can check the releases, download it and try.

It was only a compilation issue as per the comments I've seen on the releases.

**Comment by AucaCoyan** [comment]
Hello! I'll work on this issue next. I'll see what I can do!

**Comment by nadesco** [comment]
> Potential workaround: remove Espanso from the list of applications for which the _Accessibility_ permission is/isn't given, then reopen Espanso. 

Thanks, that worked! I needed to reboot my mac before the `Enable Accessibility` pane appeared though. Simply reopening Espanso after removing its _Accessibility_ permission had no effect.

**Comment by n8henrie** [comment]
Also having trouble here. Removed the old one from Accessibility, but it doesn't reappear when I run the new version (the official release, downloaded from the website). I don't see anything interesting in Console, nothing in the terminal if I launch from CLI:

```
$ open /Applications/Espanso.app
$ echo $?
0
$ ps aux | grep -i espanso | grep -v grep
$
```

If I run `/Applications/Espanso.app/Contents/MacOS/espanso daemon` it launches and can detect enable / disable events but doesn't otherwise perform any replacements.

**Comment by n8henrie** [comment]
Definitely a permissions issue.

cc @AucaCoyan 

Workaround for a one-off launch (that doesn't require a reboot):

1. Kill / stop espanso
2. Remove Espanso from System Prefs -> Privacy & Security -> Accessibility (`-` sign bottom left)
3. Install the new Espanso to `/Applications`
4. Confirm that nothing happens if you open `/Applications/Espanso.app` (future readers: hopefully this is fixed by now and you can stop here)
5. Assuming that didn't work, open a Terminal window and run: `open /Applications/Espanso.app --args launcher`
6. Accept the new accessibility permissions
7. Enjoy Espanso

Unfortunately once Espanso stops, trying to run it from the GUI (double click on app icon) *still* doesn't work. However, the CLI commands (e..g `/Applications/Espanso.app/Contents/MacOS/espanso daemon`) then prompt for accessibility permissions for the binary and subsequently running from `launchctl start com.federicoterzi.espanso` seems to work. As the launchctl commands work, most macos users should be back to a usable Espanso once they get to this point.

**Comment by FatRabidSquirrel** [comment]
> Workaround for a one-off launch (that doesn't require a reboot):

THANK YOU. Following these steps worked for me. I missed having a working Espanso for a few weeks there. Happy days.

**Comment by philippswoboda** [comment]
Any news in this regard? Since I updated to Version 2.2.7 on my Mac (via Homebrew) I'm also unable to start espanso.

**Comment by maxhoheiser** [comment]
> 5\. open /Applications/Espanso.app --args launcher

This works, but I have to follow step 2 and 5 after every restart - any updates from the def on how to solve this?

**Comment by n8henrie** [comment]
@AucaCoyan I would suggest renaming this issue for visibility. I also suspect https://github.com/espanso/espanso/issues/2475 is a duplicate and you might consider redirecting users to one issue or the other (I'd advise this one as it was opened first).

I've continued iterating on this issue and made some progress -- if I build from b6bc7cde562c30421c7f51f3b22b89ecfb12e37d
 (which requires a few patches to the vendored wx to deal with macos changes since then), I find that Espanso.app runs as expected.

So this issue seems to stem from a code change in Espanso that's happened since that commit, and I think this effectively rules out an underlying macos change as being the cause. Will see if I can run a bisect to localize the culprit commit, but the vendored Wx changes made that a little more complicated than usual.

**Comment by n8henrie** [comment]
a8539287df8dea1d373a7ab550f94db7b002211a is the culprit.

When launched from the macos app bundle, we're never making it to the "use the default `launcher` command" part, dies at `espanso/src/main.rs:447` with

```
Error {
    inner: ErrorInner {
        kind: DisplayHelpOnMissingArgumentOrSubcommand,
        context: [],
        message: Some(
```

and it then is explicitly instructed to exit.

Looks like this is mostly due to the addition of `arg_required_else_help` in that commit, which results in [`DisplayHelpOnMissingArgumentOrSubcommand`](https://docs.rs/clap/latest/clap/error/enum.ErrorKind.html#variant.DisplayHelpOnMissingArgumentOrSubcommand) if matches are empty, which they will be when launched from a macos bundle.

Time for family dinner, will whip up a PR tomorrow that should resolve this.

I bet this is another case that we could consider an integration test to pick things like this up before they become an issue.

**Comment by AucaCoyan** [comment]
Hey everyone! Nathan did a great job fixing this on #2505 and I did a pre-release to beta test it before going for the complete release.
You can try it here on https://github.com/espanso/espanso/releases/tag/v2.2.8 !

If everything goes well, I'll do a full release for all the OSes 🎉 . If it's still an issue, we will re open the issue

**Comment by dim882** [comment]
It didn't work for me originally, but eventually I got it working. I uninstalled iEspanso using CleanMyMac, and then I removed the old Espanso Accessibility items (I had two of them). Then I reinstalled Espanso and went through the wizard, and that time it worked! Thanks for fixing it!

**Comment by n8henrie** [comment]
Great to hear you got it working!

I am guessing that having to manually remove the old accessibility settings will be required across the board due to the migration of the signing key.

**Comment by archetyped** [comment]
2.2.8 beta resolved the launch issue on macOS 15.7, but also other issues as well (slow snippet output, window focus not restored after using search box, etc.).

**Comment by JSchmitt-DeepL** [comment]
After reading this bug report and the suggested fix, I can report it is not working.
macOS 26.1
espanso 2.3 (via homebrew)

## Steps taken
- Removed Accessibility permissions 
- Removed espanso
- Installed espanso
- Started setup wizard
    - Allowed to start at login (service registered)
    - Now it spawns a second instance of the setup wizard - I've closed this one in favor of the initial one.
    - Enabled Accessibility permissions (espanso not found, needed to select app manually).
    - Stuck on setup wizard, no way to move past permission request.

**Comment by n8henrie** [comment]
@JSchmitt-DeepL -- please reread the thread, we are still waiting on a 2.3.x release to get the fixes above. Until then there is a 2.2.8 -- see @AucaCoya's comment above.

**Comment by JSchmitt-DeepL** [comment]
Thanks for pointing me to the pre-release, this one works (manually installed).
Do we know roughly when this will incorporated into the main release? 2.2.8 is out for over 2 months now ::)
